vw_mqb_2010.dbc

```diff
@@ -451,8 +451,8 @@ BO_ 296 Getriebe_06: 3 Getriebe_DQ_Hybrid_MQB
  SG_ GE_Testparameter_2 : 16|8@1+ (1,0) [0|255] "" Waehlhebel_MQB
 
 BO_ 173 Getriebe_11: 8 Getriebe_DQ_Hybrid_MQB
- SG_ Getriebe_11_CRC : 0|8@1+ (1,0) [0|255] "" Gateway_MQB,Motor_Diesel_MQB,Motor_Hybrid_MQB,Motor_Otto_MQB
- SG_ Getriebe_11_BZ : 8|4@1+ (1,0) [0|15] "" Gateway_MQB,Motor_Diesel_MQB,Motor_Hybrid_MQB,Motor_Otto_MQB
+ SG_ CRC : 0|8@1+ (1,0) [0|255] "" Gateway_MQB,Motor_Diesel_MQB,Motor_Hybrid_MQB,Motor_Otto_MQB
+ SG_ COUNTERXX : 8|4@1+ (1,0) [0|15] "" Gateway_MQB,Motor_Diesel_MQB,Motor_Hybrid_MQB,Motor_Otto_MQB
```

Comment:
Typo? Also, why can't we keep the CRC/Counter signal names as they are? The suffix _CRC and _BZ seems ok. Generally, I would like to keep opendbc as close as possible to the stock dbc file.

Reply:
> Typo?

More of an ugly hack. I thought I had put a comment on that signal, but it appears to have been lost at some point. I'll add it back.

The Getriebe_11 message shows up on Extended CAN at 20Hz with the counter incrementing by five instead of one. I'm pretty sure what's happening is Getriebe_11 is sent on powertrain CAN at 100Hz and the J533 gateway is rate-limiting to conserve bandwidth. OP complains loudly about all the missing messages; that's the only reason I know this is happening! Renaming it prevents OP from doing COUNTER management on this particular message.

> Also, why can't we keep the CRC/Counter signal names as they are? The suffix _CRC and _BZ seems ok. Generally, I would like to keep opendbc as close as possible to the stock dbc file.

I'm in total agreement with you; I didn't want to change the names. Unfortunately the OP checksum code didn't really lend itself to the checksum/CRC signal having a dynamic name. It's been a couple months since I've had my head in that code so I can't call out specifics ATM.

Later today I'm going to file a PR that covers just the OP Volkswagen CRC support, because I think you'll want to look at it separately from the vehicle port. Perhaps we should hold this PR in suspense until then? I'd actually be very happy if we can arrive at a solution that keeps the DBC aligned with canonical message and signal names.

Reply:
`COUNTERXX`: yeah, reasonable hack. With the comment added is good.

`_CRC`: can't you just change https://github.com/commaai/openpilot/pull/836/files#diff-56069f2917a3366b3df2428fd6ac3202R34 so that it checks if `_CRC` is in the signal name?

I'm happy to merge this PR right away, particularly if you are ok reverting the signal checksum/counter name changes (except for the one message at 5hz with skips).

Reply:
> `_CRC`: can't you just change https://github.com/commaai/openpilot/pull/836/files#diff-56069f2917a3366b3df2428fd6ac3202R34 so that it checks if `_CRC` is in the signal name?

It might be possible in the higher level Python code, but I'm screwed once I get down to the CANPacker code where I can't regexp search on key-value pair maps:

https://github.com/commaai/openpilot/blob/799302b842198b2cccb83c601bbe79ad1eaa5e32/selfdrive/can/packer.cc#L80

https://github.com/commaai/openpilot/blob/799302b842198b2cccb83c601bbe79ad1eaa5e32/selfdrive/can/packer.cc#L109

If I knew the message name, I could concatenate that and "_CRC" or "_BZ", but I only have the address and not the name at that point, and it's not clear to me if I can look it up. That's about the point where my eyes started bleeding and I gave up. Hints welcome.

Reply:
I can help making the change to parser.cc and packer.cc to handle custom VW nomenclature. The parser side is super simple. The packer side a bit trickier, but should be easily done.
In the meanwhile, do you mind to revert the CRC/COUNTER name changes? Would like to merge these changes.